monai/handlers/iteration_metric.py

```diff
@@ -103,3 +108,20 @@ def compute(self) -> Any:
 
     def _reduce(self, scores) -> Any:
         return do_metric_reduction(scores, MetricReduction.MEAN)[0]
+
+    def attach(self, engine: Engine, name: str) -> None:
+        """
+        Attaches current metric to provided engine. On the end of engine's run,
+        `engine.state.metrics` dictionary will contain computed metric's value under provided name.
+
+        Args:
+            engine: the engine to which the metric must be attached.
+            name: the name of the metric to attach.
+
+        """
+        super().attach(engine=engine, name=name)
+        # FIXME: record engine for communication, ignite will support it in the future version soon
+        self.engine = engine
```

Comment:
this gives minor warnings: https://deepsource.io/gh/Project-MONAI/MONAI/run/1c256cfd-a167-487c-83e0-e38be1f01079/python/PYL-W0201

Reply:
fixed in the latest commit.
Thanks.